### PR TITLE
Update functions.lib.php #28795

### PR DESCRIPTION
**Pull request by carmelchas**
Promote modification from Pull request #26921 to v18 to resolve issues related to Issue breaking project and third-party dropdowns with reasonably acceptable filters that fail this test.

# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "FIX", "CLOSE", "NEW", "PERF" or "QUAL" section* (use uppercase to have the PR appears into the ChangeLog, lowercase will not appears)
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed texts with meaningful information*

# FIX|Fix #[*issue_number Short description*]
[*Long description*]

# CLOSE|Close #[*issue_number Short description*]
[*Long description*]

# NEW|New [*Short description*]
[*Long description*]

# PERF|Perf #[*issue_number Short description*]
[*Long description*]

# QUAL|Qual #[*issue_number Short description*]
[*Long description*]

### CODE REVIEW

**Comment by carmelchas** [comment]
Please help.  I really need this bug to be fixed.

File: htdocs/core/lib/functions.lib.php [hunk 1]
```diff
@@ -12017,7 +12017,7 @@ function forgeSQLFromUniversalSearchCriteria($filter, &$errorstr = '', $noand =
 	$t = preg_replace_callback('/'.$regexstring.'/i', 'dolForgeDummyCriteriaCallback', $filter);
 	$t = str_replace(array('and','or','AND','OR',' '), '', $t);		// Remove the only strings allowed between each () criteria
 	// If the string result contains something else than '()', the syntax was wrong
-	if (preg_match('/[^\(\)]/', $t)) {
+	if (!preg_match('/[^\(\)]/', $t)) {
```
Comment (eldy) [inline htdocs/core/lib/functions.lib.php:12020]: The old code looks good.
What value do you have for "$filter" at begin of function ?

**Comment by rycks** [comment]
related bug https://github.com/Dolibarr/dolibarr/pull/26921 is closed with more details about the right filter to use then that PR could be closed too i think

for me : pr to close @lvessiller-opendsi ?

**Comment by lvessiller-opendsi** [comment]
> related bug #26921 is closed with more details about the right filter to use then that PR could be closed too i think
> 
> for me : pr to close @lvessiller-opendsi ?

ok for me to close this PR

**Comment by carmelchas** [comment]
Yes.  I agree to open.

--Charles

**Comment by carmelchas** [comment]
Yes.  I agree with closing the PR.  Please excuse my previous mistaken
response.

--Charles

**Comment by lvessiller-opendsi** [comment]
@eldy
This PR can be closed.

**Comment by rycks** [comment]
ok i try to click on "close" :-)

**Pull request closed without merging**